pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -465,6 +485,82 @@ func (oe *operationExecutor) VerifyControllerAttachedVolume(
 		volumeToMount.VolumeName, "" /* podName */, verifyControllerAttachedVolumeFunc)
 }
 
+func (oe *operationExecutor) generateVolumesAreAttachedFunc(
+	attachedVolumes []AttachedVolume,
+	nodeName types.NodeName,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+
+	// volumesPerPlugin maps from a volume plugin to a list of volume specs which belong
+	// to this type of plugin
+	volumesPerPlugin := make(map[string][]*volume.Spec)
+	// volumeSpecMap maps from a volume spec to its unique volumeName which will be used
+	// when calling MarkVolumeAsDetached
+	volumeSpecMap := make(map[*volume.Spec]api.UniqueVolumeName)
```

Comment:
You are using a pointer as a key in a map. This means that the map will compare memory address for equality--not the contents of `volume.Spec`. Is that what you want? Looks like you are doing this in multiple places.

Up to you how you want to handle it. I suggest passing around `api.UniqueVolumeName` as needed and using that as the key where possible.